python/ray/tests/conftest.py

```diff
@@ -1467,6 +1467,28 @@ def random_ascii_file(request):
         yield fp
 
 
+def pytest_sessionstart(session):
+    """Called after the Session object has been created and before performing collection and entering the run test loop."""
+
+    # Shutdown Ray.
+    ray.shutdown()
+    # Kill the Ray cluster.
+    subprocess.check_call(["ray", "stop"])
+    # Delete the cluster address just in case.
+    ray._common.utils.reset_ray_address()
```

Comment:
move to a common function that can be called by both

Reply:
updated